tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,209 @@
+
+```python
+meat: str = "Ground beef"
+weight_pounds: str = 0.5
+```
+
+If we execute this file we will expect the same output as the previous one, but when we run `mypy` we will get the following message:
+
+```
+error: Incompatible types in assignment (expression has type "float", variable has type "str")
+```
+
+Did you catch the reason? The variable `weight_pounds` was defined as `str` but we were assigning a float number to it. We can start spotting bugs in every type check.
+
+I’m pretty sure you can understand this function:
+
+```python
+def make_hamburger(meat, number_of_meats):
+   return ["bread"] + [meat] * number_of_meats + ["bread"]
+
+
+print(make_hamburger("ground beef", 2))
+# ['bread', 'ground beef', 'ground beef', 'bread']
+```
+
+Let’s see how it goes with types:
+
+```python
+from typing import List
+
+
+def make_hamburger(meat: str, number_of_meats: int) -> List[str]:
+   return ["bread"] + [meat] * number_of_meats + ["bread"]
+```
+
+What happened here? We defined the function’s arguments types: string for `meat`, integer for `number_of_meats` and the return type which is a list of string values. Did you see the import at the beginning? Complex types must be imported from the `typing` library. We will cover some of them later.
+
+## Type Alias
+
+You can also define your custom type names for known structures, which is very useful for improving readability. Let’s say if we want to consider a hamburger is a list of strings, we can define a type `Hamburger` in the following way:
+
+```python
+Hamburger = List[str]
+
+def make_hamburger(meat: str, number_of_meats: int) -> Hamburger:
+   return ["bread"] + [meat] * number_of_meats + ["bread"]
+```
+
+## Callables
+
+A callable is anything you can call, using parenthesis, and possibly passing arguments. Callables are functions, classes, methods and even instances of classes can be callables if their class implements a `__call__()` method.
+
+Are you doing functional programming? No problem, the typing library inclues a type for callables which accepts a two-dimensional list like `[[argument1_type, … argumentN_type], return_type]`.
+
+```python
+from typing import Callable, Optional
+
+
+def sum_and_process(a: int, b: int, callback: Callable[[int, Optional[str]], bool]) -> bool:
+   total = a + b
+   return callback(total, None)
+
+
+def is_positive(val: int, message: Optional[str]) -> bool:
+   if message:
+       print(message)
+   return val > 0
+
+
+output = sum_and_process(5, 2, is_positive)
+print(output)
+# True
+```
+
+First of all, we are creating a function that takes two integer values, sums them up and returns the output of an incoming callback. This callback `is_positive` is defined as a function that takes one mandatory integer and one optional string. If the message exists it will print it, and always returns if the incoming value is positive. Of course, this function is not pure because of the print side-effect, but it’s just an example :)
```

Comment:
Period at end of paragraph. I think that even in a blog post we don't want to rely on emojis. Do we need to make the meaning more clear?